ibis/backends/dask/udf.py

```diff
@@ -0,0 +1,86 @@
+import itertools
+
+import dask.dataframe as dd
+import dask.dataframe.groupby as ddgb
+import pandas
+
+import ibis.client
+import ibis.expr.operations as ops
+import ibis.udf.vectorized
+from ibis.backends.pandas.udf import nullable, udf  # noqa
+
+from .dispatch import execute_node, pre_execute
+from .execution.util import make_selected_obj
+
+
+@pre_execute.register(ops.ElementWiseVectorizedUDF)
+@pre_execute.register(ops.ElementWiseVectorizedUDF, ibis.client.Client)
+def pre_execute_elementwise_udf(op, *clients, scope=None, **kwargs):
+    """Register execution rules for elementwise UDFs.
+    """
+    input_type = op.input_type
+
+    # definitions
+
+    # Define an execution rule for elementwise operations on a
+    # grouped Series
+    nargs = len(input_type)
+
+    @execute_node.register(
+        ops.ElementWiseVectorizedUDF,
+        *(itertools.repeat(ddgb.SeriesGroupBy, nargs)),
+    )
+    def execute_udf_node_groupby(op, *args, **kwargs):
+        func = op.func
+
+        # all grouping keys must be identical
+        assert len(set(tuple(arg.index) for arg in args)) == 1
+
+        # we're performing a scalar operation on grouped column, so
+        # perform the operation directly on the underlying Series
+        # and regroup after it's finished
+        args_objs = [make_selected_obj(arg) for arg in args]
+        groupings = args[0].index
+        return func(*args_objs).groupby(groupings)
```

Comment:
Does the func here takes dd.Series or pd.Series?

Reply:
dd.Series.

Reply:
This might be different from how vectorized UDF API works:

The vectorized UDF always passes a pandas data structure to the user func regardless of the backend. Therefore I believe the func here should take pd.Series instead of dd.Series.

Reply:
But then we'd have to force computation each time - defeating the point of using dask, no?

Reply:
Yeah `map_partitions` is similar to what I have in mind.

Reply:
For the spark backend, we use the API here to implement this:

https://docs.databricks.com/spark/latest/spark-sql/udf-python-pandas.html#series-to-scalar-udf

Not sure what's the best way to do in Dask, but I think either `groupby().agg()` with custom aggregation or `map_partitions` would make sense,